pypesto/ensemble/ensemble.py

```diff
@@ -367,6 +371,155 @@ def from_sample(
+                vector_tags.append((start['id'], -1))
+            else:
+                break
+
+        # print a warning if there are no vectors within the ensemble
+        if len(x_vectors) == 0:
+            raise ValueError('The ensemble does not contain any vectors. '
+                             'Either the cutoff value was too small or the '
+                             'result.optimize_result object might be empty.')
+        elif len(x_vectors) < max_size:
+            logger.info(f'The ensemble contains {len(x_vectors)} parameter '
+                        'vectors, which is less than the maximum size. If '
+                        'you want to include more vectors you can consider '
+                        'raising the cutoff value or including parameters '
+                        'from the history with the `from_history` function.')
+
+        x_vectors = np.stack(x_vectors, axis=1)
+        return Ensemble(x_vectors=x_vectors,
+                        x_names=result.problem.x_names,
+                        vector_tags=vector_tags,
+                        lower_bound=result.problem.lb_full,
+                        upper_bound=result.problem.ub_full,
+                        **kwargs)
+
+    @staticmethod
+    def from_history(
+            result: Result,
+            cutoff: float = np.inf,
+            max_size: int = np.inf,
+            max_per_start: int = np.inf,
+            **kwargs,
+    ):
+        """Construct an ensemble from the history of an optimization.
+
+        Parameters
+        ----------
+        result:
+            A pyPESTO result that contains an optimization result
+            with history recorded.
+        cutoff:
+            Exclude parameters from the optimization if the nllh
+            is higher than the `cutoff`.
+        max_size:
+            The maximum size the ensemble should be.
+        max_per_start:
+            The maximum number of vectors to be included from a
+            single optimization start.
+
+        Returns
+        -------
+        The ensemble.
+        """
+        if not result.optimize_result.list[0].history.options['trace_record']:
+            logger.warning('The optimize result has no trace. The Ensemble '
+                           'will automatically be created through '
+                           'from_optimization().')
+            return Ensemble.from_optimization_endpoints(result=result,
+                                                        cutoff=cutoff,
+                                                        max_size=max_size,
+                                                        **kwargs)
+        x_vectors = []
+        vector_tags = []
+        x_names = result.problem.x_names
+        lb = result.problem.lb_full
+        ub = result.problem.ub_full
+
+        # calculate the number of starts whose final nllh is below cutoff
+        n_starts = 0
+        for start in result.optimize_result.list:
+            if start['fval'] > cutoff:
```

Comment:
it should be. The only time this might not be (was an issue some time ago with plotting) is when you combine two different optimization runs. But I would exclude this for now tbh.